### DISCUSSION TITLE
package.json required to be present in v3

### DISCUSSION THREAD

**Discussion by raoulkent** [oas-tools/oas-tools.github.io#5] [Q&A] [answered]
Recently I've been migrating some services from v2 to v3, and of course I ran into a few small issues. But one of the more opaque ones is `packageJSON` in the configuration object.

I was wondering what it is that makes the `package.json` to be available to oas-tools? Is it due to configuration of oas-tools being possible within the `package.json` (as per [using package.json](https://oas-tools.github.io/docs/fundamentals/config#using-packagejson))?

Thanks!

**Comment by alesancor1** [comment, marked as answer]
The `packageJSON` option is used for reading configuration specified there. However, since `npm start` must always point to the project's `package.json`, I think it has much more sense to set a `serverDir` option that tells OAS Tools where the server files actually are.

This would be useful when you have a directory structure like the following:
```
L server/
L package.json
L package-lock.json
L ...
```

Then you would set `projectDir` to `./sever` which makes more sense that setting the location of the package.json, since it will always be the CWD and has no real utility.
